src/GuppyDataExplorer/ExplorerButtonGroup/index.jsx

```diff
@@ -295,14 +295,14 @@ class ExplorerButtonGroup extends React.Component {
     const caseField = this.props.guppyConfig.manifestMapping.referenceIdFieldInDataIndex;
     const caseFieldInFileIndex =
       this.props.guppyConfig.manifestMapping.referenceIdFieldInResourceIndex;
-    this.setState(prevState => ({
-      pendingManifestEntryCountRequestNumber: prevState.pendingManifestEntryCountRequestNumber + 1,
-      manifestEntryCount: 0,
-    }));
     if (this.props.buttonConfig
       && this.props.buttonConfig.buttons
       && this.props.buttonConfig.buttons.some(
         btnCfg => this.isFileButton(btnCfg) && btnCfg.enabled)) {
+      this.setState(prevState => ({
+        manifestEntryCountSemaphore: prevState.manifestEntryCountSemaphore + 1,
+        manifestEntryCount: 0,
+      }));
```

Comment:
The request for a new `manifestEntryCount` is only made inside the if statement on line 298, but the manifestEntryCountSemaphore was updated outside the if statement -- so if the if statement was never executed (this can happen on the File explorer), the semaphore would stay nonzero forever. (This caused a visual bug in the file explorer, where buttons would appear to be pending forever.)

Reply:
wow good catch!